db/sql/00_msar.sql

```diff
@@ -945,7 +954,33 @@ FROM (
     LEFT OUTER JOIN rolemembers ON r.oid = rolemembers.oid
   WHERE r.rolname NOT LIKE 'pg_%'
 ) AS role_data;
-$$ LANGUAGE sql;
+$$ LANGUAGE SQL STABLE;
+
+
+CREATE OR REPLACE FUNCTION msar.list_db_priv(db_name text) RETURNS jsonb AS $$/*
+Given a database name, returns a json array of objects with database privileges for non-inherited roles.
+
+Each returned JSON object in the array has the form:
+  {
+    "role_oid": <int>,
+    "direct" [<str>]
+  }
+*/
+WITH priv_cte AS (
+  SELECT
+    jsonb_build_object(
+      'role_oid', pgr.oid,
+      'direct',  jsonb_agg(acl.privilege_type)
+    ) AS p
+  FROM
+    pg_catalog.pg_roles AS pgr,
+    pg_catalog.pg_database AS pgd,
+    aclexplode(COALESCE(pgd.datacl, acldefault('d', pgd.datdba))) AS acl
+  WHERE pgd.oid=msar.get_db_oid(db_name) AND pgr.oid = acl.grantee AND pgr.rolname NOT LIKE 'pg_'
```

Comment:
Do we require an additional call here?

We could use `pgd.datname = db_name` instead of `pgd.oid=msar.get_db_oid(db_name)`.